gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -190,6 +192,20 @@ public GapicClass generate(GapicContext context, Service service) {
             messageTypes,
             context.restNumericEnumsEnabled());
 
+    List<MethodDefinition> methodDefinitions =
+        createClassMethods(
+            context,
+            service,
+            typeStore,
+            classMemberVarExprs,
+            callableClassMemberVarExprs,
+            protoMethodNameToDescriptorVarExprs,
+            classStatements);
+    methodDefinitions.addAll(createInvalidClassMethods(service));
```

Comment:
Can we move this HttpJson/REST specific logic to HttpJsonServiceStubClassComposer instead of in this abstract composer?

Reply:
I wasn't sure which part you were highlighting. Did you mean the entire MethodDefinition logic or just the `createInvalidClassMethods` function?